lib/rouge/lexers/mmt.rb

```diff
@@ -0,0 +1,254 @@
+        rule %r/(:)(\s*)([^\s=]+)/im do
+          groups Punctuation, Text::Whitespace, Text
+        end
+
+        rule %r/(>)([^=]+)/im do
+          groups Punctuation, Name::Variable
+        end
+
+        rule(//) { goto :moduleDefines }
+      end
+
+      state :viewHeader do
+        rule %r/\s+/im, Text::Whitespace
+
+        rule %r/(\S+)(\s*)(:)(\s*)(\S+)(\s*)(->|\u2192)(\s*)([^\u275a=]+)/im do
+          groups Name::Class, Text::Whitespace,
+                 Punctuation, Text::Whitespace,
+                 Text, Text::Whitespace,
+                 Punctuation, Text::Whitespace,
+                 Text
+          goto :moduleDefines
+        end
+      end
+
+      state :diagramHeader do
+        rule %r/\s+/im, Text::Whitespace
+
+        rule %r/(\S+)(\s*)(:)(\s*)([^\u275a=]+)/im do
+          groups Name::Variable, Text::Whitespace,
+                 Punctuation, Text::Whitespace,
+                 Name::Variable
+          goto :expression
+        end
+
+        rule %r/[^\u275a=]+/im do
+          token Name::Variable
+          goto :expression
+        end
+
+        rule %r/\u275a/im, Text, :pop!
+      end
+
+      state :structuralFeatureHeader do
+        rule %r/\s+/im, Text::Whitespace
+
+        rule %r/([^\s(\u2759\u275a:=]+)
+                (\s*)
+                (?:(\()([^)]*)(\)))?
+                (\s*)
+                (?:(:)(\s*)([^\u2758\u2759\u275a=]+))?
+                (\s*)
+                (?:(=)(\s*)([^\s:\u2759\u275a]+))?
+                (\s*)
+                (\u2759)/imx do
+          groups Name::Class,
+                 Text::Whitespace,
+                 Punctuation, Name::Variable, Punctuation,
+                 Text::Whitespace,
+                 Punctuation, Text::Whitespace, Name::Variable,
+                 Text::Whitespace,
+                 Punctuation, Text::Whitespace, Text,
+                 Text::Whitespace,
+                 Text
+          pop!
+        end
+
+        rule %r/([^\s(\u2759\u275a:=]+)
+                (\s*)
+                (?:(\()([^)]*)(\)))?
+                (\s*)
+                (?:(:)(\s*)([^\u2758\u2759\u275a=]+))?
+                (\s*)/imx do
+          groups Name::Class,
+                 Text::Whitespace,
+                 Punctuation, Name::Variable, Punctuation,
+                 Text::Whitespace,
+                 Punctuation, Text::Whitespace, Name::Variable,
+                 Text::Whitespace
+          goto :moduleDefines
+        end
+      end
+
+      state :moduleDefines do
+        rule %r/\s+/im, Text::Whitespace
+        rule %r/\u2758/im, Text
+        rule %r/#+/im, Punctuation, :notationExpression
+
+        rule %r/=/im do
+          token Punctuation
+          goto :moduleBody
+        end
+
+        rule %r/\u275a/im, Text, :pop!
+      end
+
+      state :moduleBody do
+        rule %r/\s+/im, Text::Whitespace
+        rule %r/\/T .*?(\u2759|\u275a)/im, Comment::Multiline
+        rule %r/\/\/.*?(\u2759|\u275a)/im, Comment::Multiline
```

Comment:
These rules are slightly different to the multiline comment rules in the `:root` state. Was that intentional?

Reply:
Yes, multiline comments at root are delimited by "module delimiteres" (❚) while multiline comments in modules are delimited by "declaration delimiters" (❙) *or* "module delimiters" (❚).

Reply:
Got it!